config/config.go

```diff
@@ -258,7 +284,7 @@ func (cfg *APIConfig) Validate() error {
 			return fmt.Errorf("invalid ServerURL: %w", err)
 		}
 		if cfg.TenantToken == "" {
-			log.Warn("TenantToken is empty: this device may not show up in your acount")
+			return fmt.Errorf("TenantToken (env: %s) cannot be blank", envTenantToken)
 		}
```

Comment:
@alfrunes Could we add a very explicit help message if the tenant token is the default from Azure Edge module?

Something like this?

```
if cfg.TenantToken == "<INSERT TENANT TOKEN HERE>" {
  return fmt.Errorf("Default tenant token found in env var TENANT_TOKEN - please customize tenant token in Azure IoT Edge module, or where you set environment variables / config", envTenantToken)
}
```

Reply:
A magic string? Then should we make the error message slightly more generic and include the magic string in the default configfile we ship with the tarball as well?

Reply:
> A magic string?

Yes. The point here is to help users with what looks like quite a common mistake (forgetting to edit the env var in Azure UI).

> Then should we make the error message slightly more generic and include the magic string in the default configfile we ship with the tarball as well?

Not sure? They are quite different interfaces. The reason why it's like that in the Edge module is to make it as easy as possible there. Might be different in the config file? I don't have a strong opinion, but I think the focus should be on what makes it as user friendly as possible for each interface, rather than aligning them as much as possible just to align them.

Reply:
What do you think of this 20ddb5d0178627d288c2cabebda8cca21c073024?

Reply:
@alfrunes that commit and error messages looks great to me.

I'll leave it to @kjaskiewiczz or someone else to review the rest of the code.